src/dune/bootstrap_info.ml

```diff
@@ -0,0 +1,48 @@
+open Import
+open! No_io
+
+let gen_rules sctx (exes : Dune_file.Executables.t) ~dir compile =
+  Option.iter exes.bootstrap_info ~f:(fun fname ->
+      Super_context.add_rule sctx ~loc:exes.buildable.loc ~dir
+        (Build.write_file_dyn
+           (Path.Build.relative dir fname)
+           (Build.delayed (fun () ->
+                let libs =
+                  Result.ok_exn
+                    (Lazy.force (Lib.Compile.requires_link compile))
+                in
+                let locals, externals =
+                  List.partition_map libs ~f:(fun lib ->
+                      match Lib.Local.of_lib lib with
+                      | Some x -> Left x
+                      | None -> Right lib)
+                in
+                let pr_def ppf (name, dyn) =
+                  Format.fprintf ppf "@[<2>let %s =@ %a@]" name
+                    Pp.render_ignore_tags (Dyn.pp dyn)
+                in
+                Format.asprintf "@[<v>%a@,@,%a@]@." pr_def
```

Comment:
Why do we go back to using Format for pretty printing the sources? I don't mind either way, but it seems like we've put a good amount of effort into eliminating Format wherever possible.

Reply:
Ah, yes, that's true. Changing it now

Reply:
I did that and also made the printing of Dyn.t values closer to what ocamlformat produces so that the promoted file is already well formatted